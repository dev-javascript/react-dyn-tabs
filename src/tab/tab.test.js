import React from 'react';
import {unmountComponentAtNode} from 'react-dom';
import Tab from './tab.js';
import renderer from 'react-test-renderer';
import DefaultTabInner from './defaulTabInner.js';
import Api from '../utils/api/api.js';
const tabComponent = function (props) {
  const tabProps = props.tabProps;
  tabProps.className += ' userTabComponent';
  return <DefaultTabInner {...props} tabProps={tabProps} />;
};
let container = document.createElement('div'),
  realUseContext;
const getDefaultApi = () => ({
  tabs: [
    {id: '1', title: 'tab1', tooltip: 'tab1 tootltip', closable: false},
    {id: '2', title: 'tab2', iconClass: 'ui-icon ui-icon-disk'},
    {id: '3', title: 'tab3', disable: true},
    {
      id: '4',
      title: 'tab4',
      panelComponent: function panelComponent4() {
        return <div>tab4</div>;
      },
    },
  ],
});
const setMockUseContext = (op) => {
  op = op || {};
  React.useContext = jest.fn(() => new Api({options: Object.assign({}, getDefaultApi(), op)}));
};
beforeAll(() => {
  document.body.appendChild(container);
  realUseContext = React.useContext;
});
beforeEach(() => {
  setMockUseContext();
});
afterEach(() => {
  unmountComponentAtNode(container);
  container.innerHTML = '';
  React.useContext = realUseContext;
});
afterAll(() => {
  document.body.removeChild(container);
  container = null;
});
describe('tab structure with default options : ', () => {
  test('default tab data', () => {
    setMockUseContext({
      tabs: [
        {
          title: 'tab1',
          tooltip: 'tab1',
          panelComponent: null,
          id: '1',
        },
        {
          title: 'tab2',
          tooltip: 'tab2',
          panelComponent: null,
          id: '2',
        },
      ],
    });
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('none closable tab', () => {
    setMockUseContext({
      tabs: [
        {
          title: 'tab1',
          tooltip: 'tab1',
          panelComponent: null,
          closable: false,
          iconClass: '',
          disable: false,
          id: '1',
        },
        {
          title: 'tab2',
          tooltip: 'tab2',
          panelComponent: null,
          closable: false,
          iconClass: '',
          disable: false,
          id: '2',
        },
      ],
    });
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('disabled tab', () => {
    setMockUseContext({
      tabs: [
        {
          title: 'tab1',
          tooltip: 'tab1',
          panelComponent: null,
          disable: true,
          id: '1',
        },
        {
          title: 'tab2',
          tooltip: 'tab2',
          panelComponent: null,
          disable: true,
          id: '2',
        },
      ],
    });
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('tab with provided tooltip and iconClass', () => {
    setMockUseContext({
      tabs: [
        {
          title: 'tab1',
          tooltip: 'tab1',
          panelComponent: null,
          iconClass: 'ui-icon ui-icon-home',
          id: '1',
        },
        {
          title: 'tab2',
          tooltip: 'tab2',
          panelComponent: null,
          iconClass: 'ui-icon ui-icon-book',
          id: '2',
        },
      ],
    });
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('tab structure with rtl and none accessibility options : ', () => {
  test('default tab data', () => {
    setMockUseContext({
      accessibility: false,
      direction: 'rtl',
      tabs: [
        {
          title: 'tab1',
          panelComponent: null,
          id: '1',
        },
        {
          title: 'tab2',
          panelComponent: null,
          id: '2',
        },
      ],
    });
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('none closable tab', () => {
    setMockUseContext({
      accessibility: false,
      direction: 'rtl',
      tabs: [
        {
          title: 'tab1',
          panelComponent: null,
          closable: false,
          id: '1',
        },
        {
          title: 'tab2',
          panelComponent: null,
          closable: false,
          id: '2',
        },
      ],
    });
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('disabled tab', () => {
    setMockUseContext({
      accessibility: false,
      direction: 'rtl',
      tabs: [
        {
          title: 'tab1',
          panelComponent: null,
          disable: true,
          id: '1',
        },
        {
          title: 'tab2',
          panelComponent: null,
          disable: true,
          id: '2',
        },
      ],
    });
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('tab with provided tooltip and iconClass', () => {
    setMockUseContext({
      accessibility: false,
      direction: 'rtl',
      tabs: [
        {
          title: 'tab1',
          tooltip: 'tab1',
          panelComponent: null,
          iconClass: 'ui-icon ui-icon-home',
          id: '1',
        },
        {
          title: 'tab2',
          tooltip: 'tab2',
          panelComponent: null,
          iconClass: 'ui-icon ui-icon-heart',
          id: '2',
        },
      ],
    });
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('tab structure with custom tab component : ', () => {
  test('default options', () => {
    setMockUseContext({tabComponent});
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
          <Tab id="3" selectedTabID="1"></Tab>
          <Tab id="4" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('rtl and none accessibility options', () => {
    setMockUseContext({tabComponent, direction: 'rtl', accessibility: false});
    const tree = renderer
      .create(
        <div>
          <Tab id="1" selectedTabID="1"></Tab>
          <Tab id="2" selectedTabID="1"></Tab>
          <Tab id="3" selectedTabID="1"></Tab>
          <Tab id="4" selectedTabID="1"></Tab>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
