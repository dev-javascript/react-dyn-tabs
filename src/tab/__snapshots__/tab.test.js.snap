// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab structure with custom tab component :  default options 1`] = `
<div>
  <li
    aria-controls="rc-dyn-tabs-p-1"
    aria-expanded={true}
    aria-label="tab1 tootltip"
    aria-selected={true}
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected"
    id="rc-dyn-tabs-l-1"
    onClick={[Function]}
    role="tab"
    tab-id="1"
    tabIndex={0}
    title="tab1 tootltip"
  >
    <button
      className="rc-dyn-tabs-title userTabComponent"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
    </button>
  </li>
  <li
    aria-controls="rc-dyn-tabs-p-2"
    aria-expanded={false}
    aria-label="tab2"
    aria-selected={false}
    className="rc-dyn-tabs-tab"
    id="rc-dyn-tabs-l-2"
    onClick={[Function]}
    role="tab"
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title userTabComponent"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
      <span
        className="rc-dyn-tabs-icon ui-icon ui-icon-disk"
        role="presentation"
      />
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
  <li
    aria-controls="rc-dyn-tabs-p-3"
    aria-expanded={false}
    aria-label="tab3"
    aria-selected={false}
    className="rc-dyn-tabs-tab rc-dyn-tabs-disable"
    id="rc-dyn-tabs-l-3"
    onClick={[Function]}
    role="tab"
    tab-id="3"
    tabIndex={-1}
    title="tab3"
  >
    <button
      className="rc-dyn-tabs-title userTabComponent"
      role="presentation"
      tab-id="3"
      tabIndex={-1}
    >
      tab3
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
  <li
    aria-controls="rc-dyn-tabs-p-4"
    aria-expanded={false}
    aria-label="tab4"
    aria-selected={false}
    className="rc-dyn-tabs-tab"
    id="rc-dyn-tabs-l-4"
    onClick={[Function]}
    role="tab"
    tab-id="4"
    tabIndex={-1}
    title="tab4"
  >
    <button
      className="rc-dyn-tabs-title userTabComponent"
      role="presentation"
      tab-id="4"
      tabIndex={-1}
    >
      tab4
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
</div>
`;

exports[`tab structure with custom tab component :  rtl and none accessibility options 1`] = `
<div>
  <li
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected"
    onClick={[Function]}
    tab-id="1"
    tabIndex={0}
    title="tab1 tootltip"
  >
    <button
      className="rc-dyn-tabs-title userTabComponent"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
    </button>
  </li>
  <li
    className="rc-dyn-tabs-tab"
    onClick={[Function]}
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title userTabComponent"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
      <span
        className="rc-dyn-tabs-icon ui-icon ui-icon-disk"
        role="presentation"
      />
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
  <li
    className="rc-dyn-tabs-tab rc-dyn-tabs-disable"
    onClick={[Function]}
    tab-id="3"
    tabIndex={-1}
    title="tab3"
  >
    <button
      className="rc-dyn-tabs-title userTabComponent"
      role="presentation"
      tab-id="3"
      tabIndex={-1}
    >
      tab3
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
  <li
    className="rc-dyn-tabs-tab"
    onClick={[Function]}
    tab-id="4"
    tabIndex={-1}
    title="tab4"
  >
    <button
      className="rc-dyn-tabs-title userTabComponent"
      role="presentation"
      tab-id="4"
      tabIndex={-1}
    >
      tab4
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
</div>
`;

exports[`tab structure with default options :  default tab data 1`] = `
<div>
  <li
    aria-controls="rc-dyn-tabs-p-1"
    aria-expanded={true}
    aria-label="tab1"
    aria-selected={true}
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected"
    id="rc-dyn-tabs-l-1"
    onClick={[Function]}
    role="tab"
    tab-id="1"
    tabIndex={0}
    title="tab1"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
  <li
    aria-controls="rc-dyn-tabs-p-2"
    aria-expanded={false}
    aria-label="tab2"
    aria-selected={false}
    className="rc-dyn-tabs-tab"
    id="rc-dyn-tabs-l-2"
    onClick={[Function]}
    role="tab"
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
</div>
`;

exports[`tab structure with default options :  disabled tab 1`] = `
<div>
  <li
    aria-controls="rc-dyn-tabs-p-1"
    aria-expanded={true}
    aria-label="tab1"
    aria-selected={true}
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected rc-dyn-tabs-disable"
    id="rc-dyn-tabs-l-1"
    onClick={[Function]}
    role="tab"
    tab-id="1"
    tabIndex={-1}
    title="tab1"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
  <li
    aria-controls="rc-dyn-tabs-p-2"
    aria-expanded={false}
    aria-label="tab2"
    aria-selected={false}
    className="rc-dyn-tabs-tab rc-dyn-tabs-disable"
    id="rc-dyn-tabs-l-2"
    onClick={[Function]}
    role="tab"
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
</div>
`;

exports[`tab structure with default options :  none closable tab 1`] = `
<div>
  <li
    aria-controls="rc-dyn-tabs-p-1"
    aria-expanded={true}
    aria-label="tab1"
    aria-selected={true}
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected"
    id="rc-dyn-tabs-l-1"
    onClick={[Function]}
    role="tab"
    tab-id="1"
    tabIndex={0}
    title="tab1"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
    </button>
  </li>
  <li
    aria-controls="rc-dyn-tabs-p-2"
    aria-expanded={false}
    aria-label="tab2"
    aria-selected={false}
    className="rc-dyn-tabs-tab"
    id="rc-dyn-tabs-l-2"
    onClick={[Function]}
    role="tab"
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
    </button>
  </li>
</div>
`;

exports[`tab structure with default options :  tab with provided tooltip and iconClass 1`] = `
<div>
  <li
    aria-controls="rc-dyn-tabs-p-1"
    aria-expanded={true}
    aria-label="tab1"
    aria-selected={true}
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected"
    id="rc-dyn-tabs-l-1"
    onClick={[Function]}
    role="tab"
    tab-id="1"
    tabIndex={0}
    title="tab1"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
      <span
        className="rc-dyn-tabs-icon ui-icon ui-icon-home"
        role="presentation"
      />
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
  <li
    aria-controls="rc-dyn-tabs-p-2"
    aria-expanded={false}
    aria-label="tab2"
    aria-selected={false}
    className="rc-dyn-tabs-tab"
    id="rc-dyn-tabs-l-2"
    onClick={[Function]}
    role="tab"
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
      <span
        className="rc-dyn-tabs-icon ui-icon ui-icon-book"
        role="presentation"
      />
    </button>
    <span
      className="rc-dyn-tabs-close"
      role="button"
    >
      ×
    </span>
  </li>
</div>
`;

exports[`tab structure with rtl and none accessibility options :  default tab data 1`] = `
<div>
  <li
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected"
    onClick={[Function]}
    tab-id="1"
    tabIndex={0}
    title="tab1"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
  <li
    className="rc-dyn-tabs-tab"
    onClick={[Function]}
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
</div>
`;

exports[`tab structure with rtl and none accessibility options :  disabled tab 1`] = `
<div>
  <li
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected rc-dyn-tabs-disable"
    onClick={[Function]}
    tab-id="1"
    tabIndex={-1}
    title="tab1"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
  <li
    className="rc-dyn-tabs-tab rc-dyn-tabs-disable"
    onClick={[Function]}
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
</div>
`;

exports[`tab structure with rtl and none accessibility options :  none closable tab 1`] = `
<div>
  <li
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected"
    onClick={[Function]}
    tab-id="1"
    tabIndex={0}
    title="tab1"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
    </button>
  </li>
  <li
    className="rc-dyn-tabs-tab"
    onClick={[Function]}
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
    </button>
  </li>
</div>
`;

exports[`tab structure with rtl and none accessibility options :  tab with provided tooltip and iconClass 1`] = `
<div>
  <li
    className="rc-dyn-tabs-tab rc-dyn-tabs-selected"
    onClick={[Function]}
    tab-id="1"
    tabIndex={0}
    title="tab1"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="1"
      tabIndex={-1}
    >
      tab1
      <span
        className="rc-dyn-tabs-icon ui-icon ui-icon-home"
        role="presentation"
      />
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
  <li
    className="rc-dyn-tabs-tab"
    onClick={[Function]}
    tab-id="2"
    tabIndex={-1}
    title="tab2"
  >
    <button
      className="rc-dyn-tabs-title"
      role="presentation"
      tab-id="2"
      tabIndex={-1}
    >
      tab2
      <span
        className="rc-dyn-tabs-icon ui-icon ui-icon-heart"
        role="presentation"
      />
    </button>
    <span
      className="rc-dyn-tabs-close"
    >
      ×
    </span>
  </li>
</div>
`;
